multiop.go

```diff
@@ -74,6 +84,14 @@ func (mo multiOp) Add(ops_ ...Op) Op {
 	return mo
 }
 
+func (mo multiOp) Options() Options {
+	var opts Options
+	for _, op := range mo {
+		opts.Merge(op.Options())
```

Comment:
Agree this approach is pragmatic, multiple different contexts within the same request is very much an edge case, not the default case 👍